decouple.py

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env python3
```

Comment:
Could we convert this to a `pytest` in our current test directory (skipping if `pydeps` isn't installed)? I'd prefer that since it's no longer one-more-tool to run to lint the repo (it's just a normal test), and it wouldn't clutter up the top level of the repo.

Alternatively could we move the script to the `ci/` directory, and maybe rename it to something that better indicates it's a linter script (`ci/check_decoupled.py`). The current `decouple.py` name makes it sound like the script itself decouples something.

Reply:
> Could we convert this to a pytest in our current test directory (skipping if pydeps isn't installed)?

`pydeps` is actually quite slow on ibis, so I'd rather keep it as a separate script for now.

Reply:
Any opposition to changing the name to `check_decoupled.py`/`just check_decoupled`?

Reply:
Ok, that seems like a fine compromise!